src/Netscript/NetscriptHelpers.tsx

```diff
@@ -239,6 +245,21 @@ function spawnOptions(ctx: NetscriptContext, threadOrOption: unknown): CompleteS
   return result;
 }
 
+function hostReturnOptions(returnOpts: unknown): CompleteHostReturnOptions {
+  const result: CompleteHostReturnOptions = { returnByIP: false };
+  if (typeof returnOpts !== "object" || !returnOpts) return result;
+  // Safe assertion since threadOrOption type has been narrowed to a non-null object
+  const { returnByIP } = returnOpts as Unknownify<CompleteHostReturnOptions>;
+  result.returnByIP = !!returnByIP;
+  return result;
+}
+
+/** Returns a servers hostname or IP based on the `returnByIP` field of HostReturnOptions */
+function returnServerID(server: BaseServer, returnOpts: CompleteHostReturnOptions): string {
+  const { returnByIP } = hostReturnOptions(returnOpts);
```

Comment:
This line is unneeded, because the function takes an already-resolved `CompleteHostReturnOptions`. So, this collapses down to a single ternary. It's debatable whether the single-line that remains is more clear as a helper function or not.